src/components/PressableWithDelayToggle.js

```diff
@@ -0,0 +1,145 @@
+    iconStyles: PropTypes.arrayOf(PropTypes.object),
+
+    /** Callback to be called on onPress */
+    onPress: PropTypes.func.isRequired,
+
+    /** The icon to display */
+    icon: PropTypes.func,
+
+    /** The icon to display once the pressable is pressed */
+    iconChecked: PropTypes.func,
+
+    /**
+     * Should be set to `true` if this component is being rendered inline in
+     * another `Text`. This is due to limitations in RN regarding the
+     * vertical text alignment of non-Text elements
+     */
+    inline: PropTypes.bool,
+};
+
+const defaultProps = {
+    text: '',
+    textChecked: '',
+    tooltipText: '',
+    tooltipTextChecked: '',
+    styles: [],
+    textStyles: [],
+    iconStyles: [],
+    icon: null,
+    inline: true,
+    iconChecked: Expensicons.Checkmark,
+};
+
+function PressableWithDelayToggle(props) {
+    const updatePressState = () => {
+        if (props.isDelayButtonStateComplete) {
+            return;
+        }
+        props.toggleDelayButtonState(true);
+        props.onPress();
+    };
+
+    // Due to limitations in RN regarding the vertical text alignment of non-Text elements,
+    // for elements that are supposed to be inline, we need to use a Text element instead
+    // of a Pressable
+    const PressableView = props.inline ? Text : Pressable;
+
+    return (
+        <PressableView
+            style={[styles.flexRow, ...props.styles]}
+            onPress={updatePressState}
+        >
+            <Tooltip
+                containerStyles={[styles.flexRow]}
+                text={props.isDelayButtonStateComplete ? props.tooltipTextChecked : props.tooltipText}
+            >
+                <>
+                    <Text
+                        suppressHighlighting
+                        style={[styles.mr1, ...props.textStyles]}
+                    >
+                        {props.isDelayButtonStateComplete && props.textChecked ? props.textChecked : props.text}
+                    </Text>
+                    <Pressable
+                        ref={props.innerRef}
+                        focusable
+                        accessibilityLabel={props.isDelayButtonStateComplete ? props.tooltipTextChecked : props.tooltipText}
+                        onPress={updatePressState}
+                    >
+                        {({hovered, pressed}) => (
+                            <>
+                                {props.icon && (
+                                    <Icon
+                                        src={props.isDelayButtonStateComplete ? props.iconChecked : props.icon}
+                                        fill={StyleUtils.getIconFillColor(getButtonState(hovered, pressed, props.isDelayButtonStateComplete))}
+                                        style={props.iconStyles}
+                                        width={variables.iconSizeSmall}
+                                        height={variables.iconSizeSmall}
```

Comment:
Hi, we forgot to pass `props.inline` to `<Icon />` causing https://github.com/Expensify/App/issues/22259